mage_ai/frontend/components/actions/SuggestionsList.tsx

```diff
@@ -0,0 +1,137 @@
+import api from '@api';
+import Accordion, { AccordionPanel } from '@oracle/components/Accordion';
+import Spacing from '@oracle/elements/Spacing';
+import { useMemo, useState, useEffect } from 'react';
+import SuggestionRow from './SuggestionRow';
+
+export type SuggestionsListProps = {
+  featureSet: any;
+  featureSetId: any;
+}
+
+const SuggestionsList = ({
+  featureSet,
+  featureSetId,
+}: SuggestionsListProps) => {
+
+  const suggestionsMemo = useMemo(() => (
+    (featureSet?.suggestions || [])
+  ), [
+    featureSet?.suggestions,
+  ]);
+
+  const actionsMemo = useMemo(() => (
+    (featureSet?.pipeline?.actions || [])
+  ), [
+    featureSet?.pipeline?.actions,
+  ]);
+
+  const [suggestions, setSuggestions] = useState([]);
+  const [actions, setActions] = useState(actionsMemo);
+  const [removedSuggestions, setRemovedSuggestions] = useState([]);
+
+  // initialize actions from backend on page load
+  useEffect(() => {
+    if (actionsMemo.length > 0) setActions(actionsMemo);
+  }, [
+    actionsMemo,
+  ]);
+
+  // updates suggestions and filters any removed or applied actions
+  useEffect(() => {
+    const filteredSuggestions = [...suggestionsMemo];
+    removedSuggestions.forEach(i => filteredSuggestions.splice(i, 1));
+    actions.forEach(({ i }) => filteredSuggestions.splice(i, 1));
+    setSuggestions(filteredSuggestions);
+  }, [
+    actions,
+    suggestionsMemo,
+    removedSuggestions,
+  ]);
+
+  const addAction = i => {
+    setActions(actions.concat({ i, ...suggestions[i] }));
+  };
+
+  const removeAction = i => {
+    setActions(actions.filter((x, idx) => i !== idx));
+  }
+
+  const removeSuggestion = i => {
+    setRemovedSuggestions(removedSuggestions.concat(i));
+  };
+
+  // update pipeline on backend
+  useEffect(() => {
+    if (featureSet) api.pipelines.useUpdate(featureSetId)({ actions });
+  }, [
+    actions,
+    featureSet,
+    featureSetId,
+  ]);
+
+  const actionsEl = (
+    actions.map((action, idx) => {
+      const {
+        title,
+        action_payload: {
+          action_arguments,
+        },
+      } = action;
+      const numFeatures = action_arguments.length;
+
+      return (
+        <SuggestionRow
+          idx={idx}
+          key={`${idx}-${title}`}
+          name={title}
+          numFeatures={numFeatures}
+          onClose={() => removeAction(idx)}
+          showIdx
+        />
+      );
+    })
+  );
+
+  const suggestionsEl = (
```

Comment:
You can just put this in the return statement instead of separating it out like this, espec since we've already abstracted this component from the page. We sometimes initialize components outside of the return statement when it gets too verbose, but the return statement in this component is only 3 lines.